### ISSUE TITLE
[Question] Multipart + Swagger

### ISSUE THREAD

**Issue by TomerAmirV** [Romakita/ts-express-decorators#389]
## Informations

Type | Version
---|---
Question | 4.x

****

## Description
I am trying to create a multipart endpoint (file upload).
I used the example in the docs and it seems to work fine, but I can't make the swagger doc show the file field of the request. I am also using a path param.

Here is an example

```typescript
import { Controller, PathParams, Post } from '@tsed/common';
import { MultipartFile } from '@tsed/multipartfiles';

type MulterFile = Express.Multer.File;

@Controller('/api')
export default class FileCtrl {

	@Post('/:username')
	test(@MultipartFile() myFile: MulterFile, @Required() @PathParams('username') username: string) {
		// Logic ...
	}
}
```
swagger result:
```yaml
paths:
  '/api/{username}':
    post:
      parameters:
        - in: path
          name: username
          type: string
          required: true
      consumes: []
```

Expected Result:
```yaml
paths:
  '/api/{username}':
    post:
      consumes:
         - multipart/form-data
      parameters:
        - in: path
          name: username
          type: string
          required: true
        - in: formData
          name: myFile
          type: file
          description: The file to upload.
      responses:
        '200':
          description: Success
```

Am I doing something wrong? or is this missing functionality?

Thanks!

## Acceptance criterias

- [ ] Example (log, usecase, etc...)
- [ ] Ts.ED version (bug/question)

**Comment by Romakita** [comment]
Multipart decorators didn't add information for the swagger spec.

**Comment by Romakita** [comment]
this issue can help you : #378

**Comment by TomerAmirV** [comment]
Thanks for the fast response!
this helps:
```
@Operation({consumes: ["application/x-www-form-urlencoded"]})
```

But I can't seem to set the `type` to `file`

**Comment by Romakita** [comment]
ha ok... Hum we need to improve the multipartfile module for that.

**Comment by TomerAmirV** [comment]
@Romakita do you have an estimate when it will be added? 😄

**Comment by Romakita** [comment]
Work in progress :)

**Comment by TomerAmirV** [comment]
Thanks!
Tested and it's looking good! waiting for #311 will be resolved and I'll upgrade 😄

**Comment by TomerAmirV** [comment]
Hi @Romakita!

Found two small issues,
The new code adds a creates the following swagger:
```yaml
paths:
  '/api/{username}':
    post:
      consumes:
         - multipart/form-data
      parameters:
        - in: path
          name: username
          type: string
          required: true
        - in: formData
          required: false # This fails validation and it's missing the "name" field
          type: file
```
There are two issues are:

1. It is missing the `name` field
2. It has a `required: false` field which fails swagger validation

Thanks for all your help!

**Comment by TomerAmirV** [comment]
Hi @Romakita!
Did you verify the issue? 😄

**Comment by Romakita** [comment]
Hi @TomerAmirV,

This problem is fixed with the latest release. It support only a single file right now. Multiple file isn't support by swagger ui with the swagger 2.0 spec

**Comment by TomerAmirV** [comment]
@Romakita, the second but I reported is in version 4.23.1, after the first fix

**Comment by TomerAmirV** [comment]
Hi @Romakita, any news on the second bug?

**Comment by Romakita** [comment]
ok. I didn't seen your second issue. Currently, the name doesn't exists with the Multipartfile decorator. Multipartfile use anonymous field name.

**Comment by TomerAmirV** [comment]
I think this was fixed, and it's working in version 4.25.0 🎉
